App/Logic/ExposureDetectionLogic.swift

```diff
@@ -241,10 +244,21 @@ extension Logic.ExposureDetection {
 
   /// Updates the latest processed key chunk in the `ExposureDetectionState`
   struct UpdateLatestProcessedKeyChunkIndex: AppStateUpdater {
-    let index: Int
+    let countryIndexCouple: [String: Int]
 
     func updateState(_ state: inout AppState) {
-      state.exposureDetection.latestProcessedKeyChunkIndex = self.index
+      for (key, value) in self.countryIndexCouple {
+        if key == "IT" {
+          state.exposureDetection.latestProcessedKeyChunkIndex = value
+          continue
+        }
+        for (index, countryOfInterest) in state.exposureDetection.countriesOfInterest.enumerated() {
```

Comment:
I think this can be better handled by using a dictionary in `countriesOfInterest` instead of an array. You don't need to keep countries ordered, and therefore a dictionary would be better